packages/react-composites/src/composites/localization/LocalizationProvider.tsx

```diff
@@ -65,3 +65,67 @@ export const LocalizationProvider = (props: LocalizationProviderProps): JSX.Elem
 
 /** React hook to access locale */
 export const useLocale = (): CompositeLocale => useContext(LocaleContext);
+
+interface SetLocaleContext {
+  setLocale: (locale?: string) => Promise<void>;
+}
+
+const SetLocaleContext = createContext<SetLocaleContext>({
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
+  setLocale: async (locale?: string) => {}
+});
+
+type AsyncLocalizationProviderProps = {
+  defaultLocale?: string;
+  localeLoader: (locale?: string) => Promise<CompositeLocale>;
+  children: React.ReactNode;
+};
+
+const AsyncLocalizationProvider = (props: AsyncLocalizationProviderProps): JSX.Element => {
+  const { children, defaultLocale, localeLoader } = props;
+  const [compositeLocale, setCompositeLocale] = useState<CompositeLocale>(COMPOSITE_LOCALE_EN_US);
+
+  useEffect(() => {
+    const intializeLocale = async (): Promise<void> => {
+      const compositeLocale = await localeLoader(defaultLocale);
+      setCompositeLocale(compositeLocale);
+    };
+    intializeLocale();
+  }, [defaultLocale, localeLoader]);
+
+  const state = useMemo<SetLocaleContext>(
```

Comment:
Don't see the point of `useMemo` here.

Reply:
It is not needed here. Thanks for catching that.